### DISCUSSION TITLE
Add Watch functionality to Locals

### DISCUSSION THREAD

**Discussion by corycundy** [TabularEditor/TabularEditor3#1225] [Ideas]
### Is your feature request related to a problem? Please describe

This is not a problem, but would add some efficiency when debugging. 

### Describe the solution you'd like

It would be helpful to have the Locals window have some of the same functionality that the Watch Windows has. The ability to Generate query context menu option or have the Zoom button in the Value column would be ideal so you don't have to re-type the value in the Watch Window. As an extra, if creating the context menu, the Copy Value option, similar to what the Watch Window has would be nice as well

### Goals

1. The goal is more efficiency since the current way to generate a query based on a local value it to re-type it in the Watch Window. There is currently no way to copy the expression from the Locals to the Watch. That would be a helpful along, but adding the Generate Query / Zoom buttons would be ideal.

**Comment by mlonsk** [comment]
Hi @corycundy 

I might be misunderstanding your request, but you can already right click on any variable, expression, table or column in the DAX debugger window and add it to Watch, which should allow you to Watch Locals
![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/feef719e-1bd4-433f-bfc6-191f706b652e)

![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/bb0cda02-adf9-4202-99ae-d3246fb56a2d)

Is this what you are looking for or do you have something else in mind?

**Comment by corycundy** [comment]
Ok, I didn't realize that these context menu items existed while debugging. Prior to realizing this I was looking for an enhancement to add the Zoom button (that is shown in right side of the Value column in the Watch Window) to the Value column of the Locals window. That way you can click the Zoom button (magnifying glass) within the locals and it opens a new Tab named Watch 1 in TE3 showing the query with the resulting value.

This would be used independently of the Debugger Watch Window. It would allow you to skip the step of adding something to the Watch Window and then clicking the Zoom button to open the new tab to see the more detailed results.

In other words, add the Zoom button to the right side of the Values column in Locals to be similar with the Watch windows as in the screenshot below.

![image](https://github.com/TabularEditor/TabularEditor3/assets/64218283/072c076e-33c7-4aaa-9920-4110413ec624)

Thank you though for the tip on the context menu! This helps get closer to what I was asking for.

Cory

**Reply by otykier** [reply]
Hi Cory. I agree that it would be useful to have the "Zoom" button in the Locals window, similar to how it behaves in the Watch window. We have added this request to our feature backlog 👍

**Comment by otykier** [comment]
Hi @corycundy 
Happy to let you know that we now have this functionality in [3.17.0](https://docs.tabulareditor.com/te3/other/release-notes/3_17_0.html?tabs=downloads). Moreover, we added a new "popup inspector", which is enabled by default for Locals, but you can toggle it on/off for both Locals and Watch through **Preferences > Tools > DAX Debugger**. [More information](https://docs.tabulareditor.com/te3/features/dax-debugger.html#locals).
